src/Cryptol/TypeCheck/Solver/Numeric.hs

```diff
@@ -223,6 +237,33 @@ tryCancelVar ctxt p t1 t2 =
 
 
 
+-- if (x >= 2) && x^a*x^b = x^c => a+b = c
+tryEqMulExp :: Type -> Type -> Match Solved
+tryEqMulExp x y = check x y <|> check y x
+  where 
+    check i j =
+      do  (m1,m2) <- aMul i
+          (x_1, a) <- (|^|) m1
+          (x_2, b) <- (|^|) m2
+          (x_3, c) <- (|^|) j
+          guard (x_1 == x_2 && x_2 == x_3)
+          n <- aNat x_1
+          guard (n >= 2)
+          return $ SolvedIf [ tAdd a b =#= c ]
+
+-- if (x >= 2) && x^a = x^b => a = b
```

Comment:
Same comment as `tryGeqExp`:  use `K` instead of `x` in the comment, and move the `aNat` check as early as possible.

Reply:
Updated in [110a753](https://github.com/GaloisInc/cryptol/pull/1713/commits/110a753a94506829ac11402ccf4f0067bb18bebe)